api/src/main/java/io/strimzi/api/kafka/model/KafkaMirrorMaker2MirrorSpec.java

```diff
@@ -0,0 +1,129 @@
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"sourceCluster", "targetCluster", "sourceConnector", "checkpointConnector", "heartbeatConnector", "topics", "groups"})
+@EqualsAndHashCode
+public class KafkaMirrorMaker2MirrorSpec implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private String sourceCluster;
+    private String targetCluster;
+    private KafkaConnectorSpec sourceConnector;
+    private KafkaConnectorSpec checkpointConnector;
+    private KafkaConnectorSpec heartbeatConnector;
+    private String topics;
+    private String groups;
+    private Map<String, Object> additionalProperties;
+
+    @Description("Regex of topics mirrored by the Kafka Mirror Maker 2 connectors")
+    @JsonProperty(required = true)
+    public String getTopics() {
+        return topics;
+    }
+
+    public void setTopics(String topics) {
+        this.topics = topics;
+    }
+
+    @Description("Regex of consumer groups mirrored by the Kafka Mirror Maker 2 connectors")
+    public String getGroups() {
+        return groups;
+    }
+
+    public void setGroups(String groups) {
+        this.groups = groups;
+    }
+
+    @Description("The source cluster used by the Kafka Mirror Maker 2 connectors")
+    @JsonProperty(required = true)
+    public String getSourceCluster() {
+        return sourceCluster;
+    }
+
+    public void setSourceCluster(String sourceCluster) {
+        this.sourceCluster = sourceCluster;
+    }
+
+    @Description("The target cluster used by the Kafka Mirror Maker 2 connectors")
+    @JsonProperty(required = true)
+    public String getTargetCluster() {
+        return targetCluster;
+    }
+
+    public void setTargetCluster(String targetCluster) {
+        this.targetCluster = targetCluster;
+    }
+
+    @Description("The specification of the Kafka Mirror Maker 2 source connector")
+    public KafkaConnectorSpec getSourceConnector() {
+        return sourceConnector;
+    }
+
+    public void setSourceConnector(KafkaConnectorSpec sourceConnector) {        
+        this.sourceConnector = sourceConnector;
+        if (this.sourceConnector != null) {
+            this.sourceConnector.setClassName("org.apache.kafka.connect.mirror.MirrorSourceConnector");
```

Comment:
I think it would be better to leave the classname and have the operator throw an exception (which will set a `NotReady` status condidtion) if the class name's value is non-null and incorrect. Otherwise the user will never learn that they can't set the class name, and so will carry around the misconception that setting the class name does something meaningful. You can also mention the constraints around this in the `@Description`. 

Unfortunately expression this as a JSON schema constraint (so the apiserver can enforce this constraint) isn't something the CrdGenerator can do. Though I guess you could do it abstracting a superclass for `KafkaConnectorSpec` which had all the existing properties except classname.

Reply:
I have moved the setting/checking of the connector class names to the operator and added some extra info to the `@Description`.